mapper/orm/src/main/java/org/hibernate/search/mapper/orm/event/impl/HibernateSearchEventListener.java

```diff
@@ -106,6 +106,12 @@ public void onPostDelete(PostDeleteEvent event) {
 
 		Object providedId = typeContext.toIndexingPlanProvidedId( event.getId() );
 		plan.delete( providedId, null, entity );
+
+		BitSet dirtyAssociationPaths = typeContext.dirtyContainingAssociationFilter().all();
+
+		if ( dirtyAssociationPaths != null ) {
+			plan.updateAssociationInverseSide( dirtyAssociationPaths, null, event.getDeletedState() );
+		}
```

Comment:
I'm not sure of the implications of this... I'll try to have a closer look tomorrow.

Reply:
Okay so this was copy-pasted from `onPostInsert`/`onPostUpdate`...
It should be at least as safe as it is there, then.

I wonder what happens when the target of the assocation in `deletedState` has also been deleted, but to be fair if that's a problem, it's probably a problem in the `oldState` of `onPostUpdate` too. So, out of scope here.

I'd simply suggest you add a comment just like we did on  `onPostInsert`/`onPostUpdate` :)